combiner/combiner.go

```diff
@@ -0,0 +1,208 @@
+package combiner
+
+import (
+	"runtime"
+	"sync"
+
+	"github.com/philippgille/gokv"
+)
+
+// Check we satisfy interface.
+var _ gokv.Store = &Store{}
+
+// Store is a `gokv.Store` implementation that forwards its
+// calls to other `gokv.Store`s.
+type Store struct {
+	backends []gokv.Store
+	parallel bool
+}
+
+// A helper type to check for errors when creating `gokv.Store`s.
+type Backend struct {
+	store gokv.Store
+	err   error
+}
+
+// The options for this `gokv.Store` implementation.
+type Options struct {
+	Backends []Backend
+	Parallel bool
```

Comment:
This is good for the first version of a combiner. But in the future it would be awesome to have "strategies" for setting, getting and deleting values.

For example:
- When setting, make it a blocking call until the quickest store has a value, then return with the function but keep setting to the other store in the background, and call a given callback function (or notify me via given channel) when an error occurs in the background goroutine.
- When setting, block all calls and only return when all are finished
- When getting, I only want the fastest answer, cancel all other operations (we might need to introduce a context object for this)
- When getting and using 3 stores, wait for all operations, then compare their values and return an error if one of them is different
- When getting and using 3 stores, wait for all operations, then compare their values and return an error if _the majority_ of them is different. If the majority is the same then "trust" this majority and call `Set` in the background to the minority of Stores with "wrong" values and overwrite them with the "correct" ones
- ...

I'd love to discuss the usefulness of these strategies. Maybe that's something for a new enhancement issue.